tests/context-menu.spec.ts

```diff
@@ -1,4 +1,262 @@
-import { test, expect } from '@playwright/test';
+import { test, expect, Page } from '@playwright/test';
+import {
+  makeTableCommand,
+  pasteInMonaco,
+  TableTypes,
+  waitForLoadingDone,
+  generateVarName,
+} from './utils';
+
+test.describe.configure({ mode: 'serial' });
+
+async function openAdvancedFilters(page: Page) {
+  await page
+    .locator('.iris-grid .grid-wrapper')
+    .click({ button: 'right', position: { x: 20, y: 20 } });
+  await page.getByRole('button', { name: 'Advanced Filters' }).click();
+}
+
+async function moveMouseAwayFromTable(page: Page) {
+  await page.mouse.move(0, 0);
+  await page.mouse.click(0, 0);
+}
+
+const tableName = generateVarName('t');
+test.beforeEach(async ({ page }) => {
+  await page.goto('');
+
+  const consoleInput = page.locator('.console-input');
+
+  const command = makeTableCommand(tableName, TableTypes.AllTypes);
+
+  await pasteInMonaco(consoleInput, command);
+  await page.keyboard.press('Enter');
+
+  // Wait for the panel to show
+  await expect(page.locator('.iris-grid-panel')).toHaveCount(1);
+
+  // Wait until it's done loading
+  await expect(page.locator('.iris-grid-panel .loading-spinner')).toHaveCount(
+    0
+  );
+
+  // Model is loaded, need to make sure table data is also loaded
+  await waitForLoadingDone(page);
+
+  const tableOperationsMenu = page.locator(
+    'data-testid=btn-iris-grid-settings-button-table'
+  );
+  await tableOperationsMenu.click();
+
+  // Wait for Table Options menu to show
+  await expect(page.locator('.table-sidebar')).toHaveCount(1);
```

Comment:
This is exactly the same setup as in `table-operations.spec.ts`. We could refactor this into a utility method used by both, but I think in this case I would rather just move these table specific context menu tests to `table-operations.spec.ts` (since we're really testing the table operations here, we're just accessing them from the context menu).